### PR TITLE
Adds buffered configurers

### PR DESCRIPTION
**Pull request by fwh-dc**
Adds private key buffer, certificate buffer and certificate chain buffer configurers

### CODE REVIEW

**Comment by fwh-dc** [comment]
Hi @lganzzzo 

Hope you have time for this one as well 🙏

**Comment by lganzzzo** [comment]
Hello @fwh-dc ,
Looks good!

Please add a test using buffered certs. - you can just use an existing [FullTest](https://github.com/oatpp/oatpp-openssl/blob/master/test/oatpp-openssl/FullTest.hpp)
Just add an option there so it could be run both with file certs and buffered cets.

Regards,
Leonid

**Comment by fwh-dc** [comment]
Good catch @lganzzzo I've added it now.

File: test/oatpp-openssl/FullTest.cpp [hunk 1]
```diff
@@ -78,13 +83,40 @@ class TestComponent {
...
+    OATPP_LOGD("oatpp::openssl::Config", "pem='%s' loaded from %s", CERT_PEM_PATH, m_useBufferedCertAndPrivateKey ? "memory" : "file");
+    OATPP_LOGD("oatpp::openssl::Config", "crt='%s' loaded from %s", CERT_CRT_PATH, m_useBufferedCertAndPrivateKey ? "memory" : "file");
     OATPP_LOGD("oatpp::openssl::Config", "dh_params='%s'", CERT_DH_PARAMS_PATH);
+    std::shared_ptr<oatpp::openssl::Config> config;
 
-    auto config = oatpp::openssl::Config::createDefaultServerConfigShared(CERT_CRT_PATH, CERT_PEM_PATH);
+    if(m_useBufferedCertAndPrivateKey) {
+      config = oatpp::openssl::Config::createShared();
+
+      std::ifstream certFile(CERT_CRT_PATH, std::ios::binary | std::ios::ate);
```
Comment (lganzzzo) [inline test/oatpp-openssl/FullTest.cpp]: Please use `oatpp::String::loadFromFile()` instead:

```cpp
auto certBuffer = oatpp::String::loadFromFile(CERT_CRT_PATH);
auto privateKeyBuffer = oatpp::String::loadFromFile(CERT_PEM_PATH);

config->addContextConfigurer(std::make_shared<oatpp::openssl::configurer::PrivateKeyBuffer>(privateKeyBuffer->data(), privateKeyBuffer->size()));
config->addContextConfigurer(std::make_shared<oatpp::openssl::configurer::CertificateBuffer(certBuffer->data(), certBuffer->size());
```

**Comment by lganzzzo** [comment]
Hello @fwh-dc ,

Looks good!
Please simplify cert loading as described above and it's ready to merge.

Regards,
Leonid

**Comment by fwh-dc** [comment]
Nice, didn't notice you had the functionality.

**Comment by lganzzzo** [comment]
Great work!
Merged!

**Pull request merged**